storm-core/src/jvm/storm/trident/topology/TridentTopologyBuilder.java

```diff
@@ -248,8 +250,9 @@ public StormTopology buildTopology() {
     }
     
     private void markBatchGroups(String component, Map<String, String> batchGroups) {
-        for(String stream: batchGroups.keySet()) {
-            _batchIds.put(new GlobalStreamId(component, stream), batchGroups.get(stream));
+        for(Map.Entry<String, String> entry: batchGroups.entrySet()) {
+            String stream = entry.getKey();
+            _batchIds.put(new GlobalStreamId(component, stream), entry.getValue());
```

Comment:
Unless the variable is created to make the code easier to read, this could be simplified to 
_batchIds.put(new GlobalStreamId(component, entry.getKey()), entry.getValue());